packages/tpetra/core/test/Map/Map_getMinGlobalIndex.cpp

```diff
@@ -0,0 +1,212 @@
+// put these into test_same_as and test_is_compatible
+
+namespace {
+
+  using Teuchos::null;
+
+  using Tpetra::TestingUtilities::getNode;
+  using Tpetra::TestingUtilities::getDefaultComm;
+
+  using Tpetra::createUniformContigMapWithNode;
+
+  using Teuchos::Array;
+  using Teuchos::ArrayView;
+  using Teuchos::as;
+  using Teuchos::RCP;
+  using Teuchos::arcp;
+  using Teuchos::rcp;
+  using Teuchos::outArg;
+  using Teuchos::Tuple;
+  using Teuchos::tuple;
+  using Tpetra::Map;
+  using Tpetra::global_size_t;
+  using Tpetra::DefaultPlatform;
+  using std::sort;
+  using std::find;
+  using Teuchos::broadcast;
+  using Teuchos::OrdinalTraits;
+  using Teuchos::Comm;
+  using std::endl;
+
+  ////
+  TEUCHOS_UNIT_TEST_TEMPLATE_2_DECL( Map, getMinGlobalIndex_nonuniform, LO, GO )
+  {
+    typedef Map<LO,GO> Map;
+
+    out << "Test: Map, getMinGlobalIndex" << std::endl;
+    Teuchos::OSTab tab0 (out);
+
+    // create a comm
+    RCP<const Comm<int> > comm = getDefaultComm();
+    const int numRanks = comm->getSize();
+    const int myRank = comm->getRank();
+    // create a contiguous uniform distributed map with numLocal entries per node
+    const size_t        numLocalRef = 5;
+    const size_t        numLocal    = (myRank == 0) ? 0 : numLocalRef;
+    const global_size_t numGlobal   = (numRanks - 1)*numLocalRef;
+    const GO indexBase = 0;
+    //
+
+    Map map (numGlobal, numLocal, indexBase, comm);
+    // create data to check validity of the map
+    const GO myMinGID =
+      (myRank == 0) ? std::numeric_limits<GO>::max() : (myRank - 1)*numLocal;
+    const GO myMaxGID =
+      (myRank == 0) ? std::numeric_limits<GO>::lowest() : myRank*numLocal - 1;
+    GO minAllGIDs, maxAllGIDs;
+    Teuchos::reduceAll<LO, GO>(*comm, Teuchos::REDUCE_MIN, myMinGID, outArg(minAllGIDs));
+    Teuchos::reduceAll<LO, GO>(*comm, Teuchos::REDUCE_MAX, myMaxGID, outArg(maxAllGIDs));
+    std::cout << "p=" << myRank << " | map.getNodeElementList():" <<map.getNodeElementList()
+              << std::endl;
+    std::cout << "p=" << myRank << " | map.getMinGlobalIndex(): "
+              << map.getMinGlobalIndex() << ", myMinGID=" << myMinGID << std::endl;
+    std::cout << "p=" << myRank << " | map.getMinAllGlobalIndex(): "
+              << map.getMinAllGlobalIndex() << ", minAllGID=" << minAllGIDs << std::endl;
+    std::cout << "p=" << myRank << " | map.getMaxGlobalIndex(): "
+              << map.getMaxGlobalIndex() << ", myMaxGID=" << myMaxGID << std::endl;
+    std::cout << "p=" << myRank << " | map.getMaxAllGlobalIndex(): "
+              << map.getMaxAllGlobalIndex() << ", maxAllGID=" << maxAllGIDs << std::endl;
+    //
+    TEST_EQUALITY(map.getGlobalNumElements(), numGlobal);
+    TEST_EQUALITY(map.getNodeNumElements(), numLocal);
+    TEST_EQUALITY(map.getIndexBase(), indexBase);
+    TEST_EQUALITY_CONST(map.getMinLocalIndex(), 0);
+    TEST_EQUALITY(map.getMaxLocalIndex(), numLocal - 1);
```

Comment:
The issue doesn't say anything about the behavior of `getMinLocalIndex` or `getMaxLocalIndex`, so the test should not test those methods.

Reply:
What I mean is that your test overconstrains the solution to the issue.  Currently, getMinLocalIndex always returns 0, even if the calling process has no indices.  You might like to change that behavior in the future, so that users could write a loop like this:
```
for (LO lid = map.getMinLocalIndex(); lid <= getMaxLocalIndex(); ++lid) { /* ... */ }
```
However, making that work is a separate issue.

Reply:
OK, I removed these two checks (again, lazy copy paste). I would be indeed a different issue although quite related to this one. We should make a note to discuss that later.